[lldb] fix python extension debug suffix on Win

amordo opened the pull request:
ae389b2450bd604a3f3bbe5b09b333b2d99801dd change doesn't cover "_d" suffix for Debug build on Windows.

Fixed #87381.

DavidSpickett commented:
The best docs I can find are https://peps.python.org/pep-0720/:
"Interpreter-specific Python extension (native module) suffix — generally defined as .{SOABI}.{SHLIB_SUFFIX}."

Which is less than helpful. https://peps.python.org/pep-3149/ mentions Windows uses a different format.

But anyway, just curious exactly what your interpreter returns.

DavidSpickett commented:
So this PR fixes debug lldb -> release Python, but if I configured against a debug Python then the suffix would be `_d_d`.

Debug lldb/release Python is a valid config I think, and likely what 99% of people want anyway. So can we add a check whether the suffix already has `_d` in it, so that debug lldb/debug python also works?

DavidSpickett commented on lldb/CMakeLists.txt:
```diff
@@ -75,6 +75,12 @@ if (LLDB_ENABLE_PYTHON)
   endforeach()
+  if(WIN32 AND CMAKE_BUILD_TYPE STREQUAL Debug)
+    string(SUBSTRING ${LLDB_PYTHON_EXT_SUFFIX} 0 2 FIRST_2_CHARS)
+    if(NOT FIRST_2_CHARS STREQUAL "_d")
```
Add some comments to explain why we have this check, something like:
```
# If we are using a debug Python, the expected suffix will already have "_d".
if (...)
  # We are using a release Python, but the lldb extension will be built with the "_d" suffix.
```

amordo committed:
[lldb] check python extension debug suffix on Win

ae389b2450bd604a3f3bbe5b09b333b2d99801dd relies on
LLDB_PYTHON_EXT_SUFFIX of Python3_EXECUTABLE for naming the result lldb
python lib.

Debug python is used in Debug LLDB on Windows, so the result lib name
requires "_d". LLDB_PYTHON_EXT_SUFFIX doesn't start with "_d" if
Python3_EXECUTABLE wasn't set to Debug executable explicitly. Perhaps a
better solution can be found after solving "debug python executable is
not currently handled" issue
(https://gitlab.kitware.com/cmake/cmake/-/issues/25874#note_1506658).

DavidSpickett reviewed (approved):
LGTM

I can merge this for you if you don't have commit access, otherwise go ahead and merge it yourself.
